core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -119,8 +126,6 @@ object SizeEstimator extends Logging {
       }
     }
     pointerSize = if (is64bit && !isCompressedOops) 8 else 4
-    classInfos.clear()
-    classInfos.put(classOf[Object], new ClassInfo(objectSize, Nil))
```

Comment:
The map is expected to contain class info from the current execution; not from past runs (which might or might not be relevant - increasing the map size).